### PR TITLE
Remove BLOCK_OBSERVATION keyword

### PR DESCRIPTION
**Pull request by dafeda**
## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by dafeda** [comment]
test ert please

File: tests/ert_tests/data/test_loader.py [hunk 1]
```diff
@@ -285,9 +286,7 @@ def test_different_data_key(facade):
 
 
 @deprecation.fail_if_not_removed
-@pytest.mark.parametrize(
-    "func", [loader.load_general_data, loader.load_summary_data, loader.load_block_data]
-)
+@pytest.mark.parametrize("func", [loader.load_general_data, loader.load_summary_data])
 def test_deprecated_entry_points(facade, monkeypatch, func):
```
Comment (oyvindeide) [inline tests/ert_tests/data/test_loader.py:290]: Think you are removing the deprecated entry points, so this whole test can be deleted too?

**Review by oyvindeide — COMMENTED** [review]
This looks really nice! Did you check what happens when someone runs a config with `BLOCK_OBSERVATION` after this?

File: tests/ert_tests/data/test_loader.py (hunk 1 above)
Comment (dafeda) [reply tests/ert_tests/data/test_loader.py:255]: Thanks, fixed.

**Comment by dafeda** [comment]
> This looks really nice! Did you check what happens when (if) someone runs a config with `BLOCK_OBSERVATION` after this?

I added 

```
BLOCK_OBSERVATION RFT_2006
{
   FIELD  = PRESSURE;
   DATE   = 2006-10-22;
   SOURCE = SUMMARY;

   OBS P1 { I = 1;  J = 1;  K = 1;   VALUE = 100;  ERROR = 5; };
   OBS P2 { I = 2;  J = 2;  K = 1;   VALUE = 101;  ERROR = 5; };
   OBS P3 { I = 2;  J = 3;  K = 1;   VALUE = 102;  ERROR = 5; };
};
```

to `snake_oil_field/snake_oil.ert` and ran `ert gui snake_oil.ert`, which throws the following.

```
2022-08-24 11:59:38,108 - res.config - WARNING - ** Warning keyword: BLOCK_OBSERVATION not recognized when parsing: snake_oil.ert ---
2022-08-24 11:59:38,108 - res.config - WARNING - ** Warning keyword: { not recognized when parsing: snake_oil.ert ---
2022-08-24 11:59:38,108 - res.config - WARNING - ** Warning keyword: DATE not recognized when parsing: snake_oil.ert ---
2022-08-24 11:59:38,108 - res.config - WARNING - ** Warning keyword: SOURCE not recognized when parsing: snake_oil.ert ---
2022-08-24 11:59:38,108 - res.config - WARNING - ** Warning keyword: OBS not recognized when parsing: snake_oil.ert ---
2022-08-24 11:59:38,108 - res.config - WARNING - ** Warning keyword: OBS not recognized when parsing: snake_oil.ert ---
2022-08-24 11:59:38,108 - res.config - WARNING - ** Warning keyword: OBS not recognized when parsing: snake_oil.ert ---
2022-08-24 11:59:38,108 - res.config - WARNING - ** Warning keyword: }; not recognized when parsing: snake_oil.ert ---

Error message: ensemble_config_init_FIELD: field type: PRESSURE; is not recognized
```

**Review by oyvindeide — APPROVED** [review]
LGTM!

**Pull request merged**